datasync/chaindatafetcher/kafka/consumer.go

```diff
@@ -177,13 +189,13 @@ func insertSegment(newSegment *Segment, buffer [][]*Segment) ([][]*Segment, erro
 		if numBuffered > 0 && bufferedSegments[0].key == newSegment.key {
 			// there is a missing segment which should not exist.
 			if newSegment.index > uint64(numBuffered) {
-				logger.Error("there may be a missing segment", "numBuffered", numBuffered, "newSegment", newSegment)
+				Logger.Printf("there may be a missing segment [numBuffered: %d, newSegment: %s]\n", numBuffered, newSegment.String())
 				return buffer, errors.New(missingSegmentErrorMsg)
 			}
 
 			// the segment is already inserted to buffer.
 			if newSegment.index < uint64(numBuffered) {
-				logger.Warn("the message is duplicated", "newSegment", newSegment)
+				Logger.Printf("the message is duplicated [newSegment: %s]\n", newSegment.String())
```

Comment:
Can we set an error level? I think this message will be printed too much when multi-producers are used.

Reply:
I wanted to. I decided to use sarama stdlogger since the client must use sarama package.
Or, we may create a logger with log levels and I think it makes the user confused. Any other thought?

Reply:
Also, as we have producer id, the duplication message may not be printed too much. The scenarios which I can think of are
1. Kafka sends messages twice in an abnormal situation.
2. Fetching and RangeFetching happens at the same moment.
3. ...